src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Operators.vb

```diff
@@ -1123,13 +1110,7 @@ Namespace Microsoft.VisualBasic.CompilerServices
         End Function
 
         Private Shared Function NegateDecimal(ByVal operand As Decimal) As Object
-            'Using try/catch instead of check with MinValue since the overflow case should be very rare
-            'and a compare would be a big cost for the normal case.
-            Try
-                Return -operand
-            Catch ex As OverflowException
-                Return -CDbl(operand)
-            End Try
+            Return -operand
```

Comment:
We have tests negating decimal.MinValue - this does not cause an overflow exception

```vb
yield return new object[] { decimal.MinValue, decimal.Parse("79228162514264337593543950335") };
```

Reply:
Can we add tests with MaxValue as well (applicable to others)?

---
In reply to: [204927252](https://github.com/dotnet/corefx/pull/31320#discussion_r204927252) [](ancestors = 204927252)